modules/dists/BlockDist.chpl

```diff
@@ -962,16 +986,19 @@ proc BlockDom.dsiLocalSlice(param stridable: bool, ranges) {
 }
 
 proc BlockDom.setup() {
-    coforall localeIdx in dist.targetLocDom do {
-      on dist.targetLocales(localeIdx) do
-        locDoms(localeIdx).myBlock = dist.getChunk(whole, localeIdx);
+  coforall (localeIdx, loc, locDomsElt)
+           in zip(dist.targetLocDom, dist.targetLocales, locDoms) {
+    on loc {
+      locDomsElt.myBlock = dist.getChunk(whole, localeIdx);
     }
+  }
 }
 
 override proc BlockDom.dsiDestroyDom() {
-  coforall localeIdx in dist.targetLocDom do {
-    on locDoms(localeIdx) do
-      delete locDoms(localeIdx);
+  coforall (localeIdx, locDomsElt) in zip(dist.targetLocDom, locDoms) {
```

Comment:
drop `localeIdx`?

In `dsiAssign`, you choose to zip `targetLocales`, too, that could avoid `on locDomElts` here. I wonder if you have a sense of which one makes more sense from performance standpoint. My hunch would be that this should also iterate over targetLocales and do `on loc`. I don't know much about how `coforall ... zip` is implemented though.

Even if it has overheads, replacing `dist.targetLocDom` with `dist.targetLocales` seems like the right thing to do here.

Reply:
I'll give it a try but I don't have any reason to expect it will change performance.

Reply:
FWIW, as long as `localeIdx` is gone, I am happy. No need to do another perf test just for its sake.